pkg/controller/klusterletaddon/klusterlet_addon_controller.go

```diff
@@ -381,3 +394,16 @@ func IsCRDManfestWorkAvailable(manifestWork *manifestworkv1.ManifestWork) bool {
 
 	return false
 }
+
+func setManagedClusterPolicyControllerLabel(managedCluster *managedclusterv1.ManagedCluster, klusterletAddonConfig *agentv1.KlusterletAddonConfig, client client.Client) error {
+	if klusterletAddonConfig.DeletionTimestamp != nil ||
```

Comment:
why not check the existence of managedclusteraddon?

Reply:
It will be less performant to check on the managedclusteraddon as I will have to reread it to find out if it exists or not.

Reply:
@qiujian16 I don't find in the code where the managedClusterAddOn is deleted, it seems the sync (https://github.com/open-cluster-management/klusterlet-addon-controller/blob/545efd65a42c7c2b7f931e88a70008daff8bd136/pkg/controller/klusterletaddon/klusterlet_addon_crs.go#L120)  only update the object but not delete it.

Reply:
ok it should be done somewhere because I tested and it works, the managedclusteraddon is effectively deleted.